app/src/main/java/com/example/bigowlapp/utils/AuthenticatorByPhoneNumber.java

```diff
@@ -0,0 +1,65 @@
+package com.example.bigowlapp.utils;
+
+import android.annotation.SuppressLint;
+import android.content.Context;
+import android.telephony.TelephonyManager;
+import android.widget.Toast;
+
+import com.example.bigowlapp.model.Attendance;
+import com.example.bigowlapp.model.AuthByPhoneNumberFailure;
+import com.example.bigowlapp.model.Schedule;
+import com.example.bigowlapp.model.User;
+import com.example.bigowlapp.model.UserScheduleResponse;
+import com.example.bigowlapp.repository.AuthRepository;
+import com.example.bigowlapp.repository.RepositoryFacade;
+import com.google.firebase.Timestamp;
+import com.google.firebase.installations.FirebaseInstallations;
+
+public class AuthenticatorByPhoneNumber {
+
+    private String devicePhoneNumber;
+    private String currentUserPhoneNumber;
+    private final Context context;
+    private final AuthRepository authRepository = new AuthRepository();
+    private RepositoryFacade repositoryFacade = RepositoryFacade.getInstance();
+
+    public AuthenticatorByPhoneNumber(Context context) {
+        this.context = context;
+    }
+
+    @SuppressLint("MissingPermission")
+    // Permission was already granted after sign in step to be able to proceed
+    public void authenticate(String scheduleId) {
+        TelephonyManager telephonyManager = (TelephonyManager) context.getSystemService(Context.TELEPHONY_SERVICE);
+        devicePhoneNumber = telephonyManager.getLine1Number();
+        repositoryFacade.getScheduleRepository().getDocumentByUid(scheduleId, Schedule.class)
+                .observeForever(schedule -> {
+                    repositoryFacade.getUserRepository()
+                            .getDocumentByUid(repositoryFacade.getAuthRepository().getCurrentUser().getUid(), User.class)
+                            .observeForever(user -> {
+                                currentUserPhoneNumber = user.getPhoneNumber();
+                                UserScheduleResponse userScheduleResponse = schedule.getUserScheduleResponseMap()
+                                        .get(repositoryFacade.getAuthRepository().getCurrentUser().getUid());
+                                Attendance attendance = userScheduleResponse.getAttendance();
+                                if (currentUserPhoneNumber.equalsIgnoreCase("+" + devicePhoneNumber)) {
```

Comment:
Cause of Case 1.

The problem with this method is that it assumes that the phone number received from Android does not have a "+" at the beginning of the number.

However, in some devices, there would be a "+" and sometimes there wouldn't be one. I believe it all depends on the OS type, phone provider, and the phone type. This is shown in the image.

![phones](https://user-images.githubusercontent.com/23643192/111890019-230f6300-89bc-11eb-9d20-a9c0fe0ed94e.png)

As a solution, I recommend using a phone number formatter.
This is the method we should use, and how it is used.
https://github.com/Landry333/Big-Owl/blob/0573cc53f0d3fe3f335a3c5ca9aa8ee9b73294ed/app/src/main/java/com/example/bigowlapp/activity/SignUpPageActivity.java#L100-L106
https://github.com/Landry333/Big-Owl/blob/0573cc53f0d3fe3f335a3c5ca9aa8ee9b73294ed/app/src/main/java/com/example/bigowlapp/activity/SignUpPageActivity.java#L57

To prevent any duplicated code, I recommend extracting it into a class instead of copy-pasting the same method.

Reply:
ok, I agree. That seems a good idea